tests/test_throttling.py

```diff
@@ -159,6 +159,27 @@ def test_request_throttling_multiple_throttles(self):
         assert response.status_code == 429
         assert int(response['retry-after']) == 58
 
+    def test_handle_negative_throttle_value(self):
```

Comment:
Minor nitpick. I might change this to `test_throttle_rate_reduction` or something similar. The fact that the throttle value went negative is just an implementation detail that isn't directly being tested here.

Reply:
Indeed, I went with `test_throttle_rate_change_negative` in  8e6ee98